app/models/alchemical_property.rb

```diff
@@ -6,13 +6,25 @@ class AlchemicalProperty < ApplicationRecord
   has_many :canonical_ingredients_alchemical_properties,
            dependent: :destroy,
            class_name: 'Canonical::IngredientsAlchemicalProperty'
-  has_many :canonical_ingredients, through: :canonical_ingredients_alchemical_properties
+  has_many :canonical_ingredients,
+           through: :canonical_ingredients_alchemical_properties,
+           source: :ingredient
 
   has_many :canonical_potions_alchemical_properties,
            dependent: :destroy,
            class_name: 'Canonical::PotionsAlchemicalProperty'
   has_many :canonical_potions, through: :canonical_potions_alchemical_properties, source: :potion
 
+  has_many :ingredients_alchemical_properties,
```

Comment:
Apparently I neglected to add this code when I created the non-canonical `Ingredient` model 😅